src/plugins/pcr_primer_design_for_dna_assembly/src/PCRPrimerProductTable.cpp

```diff
@@ -0,0 +1,92 @@
+/**
+ * UGENE - Integrated Bioinformatics Tools.
+ * Copyright (C) 2008-2021 UniPro <ugene@unipro.ru>
+ * http://ugene.net
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston,
+ * MA 02110-1301, USA.
+ */
+
+#include <U2Core/AnnotationSelection.h>
+#include <U2Core/AnnotationTableObject.h>
+#include <U2Core/DNASequenceSelection.h>
+#include <U2Core/U2SafePoints.h>
+
+#include <U2View/ADVSequenceObjectContext.h>
+#include <U2View/AnnotatedDNAView.h>
+
+#include "PCRPrimerProductTable.h"
+#include "src/tasks/PCRPrimerDesignForDNAAssemblyTask.h"
+
+namespace U2 {
+
+ PCRPrimerProductTable::PCRPrimerProductTable(QWidget *parent)
+    : QTableWidget(parent) {
+     setColumnCount(2);
+     setRowCount(8);
+     setHorizontalHeaderLabels(QStringList() << tr("Fragment") << tr("Region"));
+     setSelectionBehavior(QAbstractItemView::SelectRows);
+     setSelectionMode(QAbstractItemView::SingleSelection);
+     connect(selectionModel(), SIGNAL(selectionChanged(const QItemSelection &, const QItemSelection &)), SLOT(sl_selectionChanged()));
+     connect(this, SIGNAL(clicked(const QModelIndex &)), SLOT(sl_selectionChanged()));
+ }
+
+void PCRPrimerProductTable::setCurrentProducts(const QList<U2Region> &currentProducts, AnnotatedDNAView *_associatedView) {
+    SAFE_POINT(currentProducts.size() == 8, "Should be 8 results", );
```

Comment:
And again - it could be less than eight rows

Reply:
As I suppose, the task always returns 8 regions, some or all of them can be empty - then they are not included in the result. Since I was not provided with information about the format of the outputted results of the task, I assumed that there are always 8 results.

Reply:
OK, it this case, create a `static constexprt` variable to avoid magic numbers